src/libs/actions/Report.js

```diff
@@ -611,14 +611,36 @@ function deleteReportComment(reportID, reportAction) {
         text: '',
         isEdited: true,
     }];
+
+    const isDeletedPendingAdd = reportAction.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD;
+
     const optimisticReportActions = {
         [sequenceNumber]: {
             pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
             previousMessage: reportAction.message,
             message: deletedMessage,
+
+            // If we're trying to delete a report action that's still pending add,
+            // then we must be offline and the report action must have been added in the same offline session.
+            // This hides any reportActions that are added after AddComment fires
+            isDeletedPendingAdd,
         },
     };
 
+    // If we're trying to delete a report action that's still pending add,
+    // then we must be offline and the optmistic report action must have been added in the same offline session.
+    // Therefore, let's just hide it for the time being.
+    if (isDeletedPendingAdd) {
+        // Optimistic report actions are saved into the clientID, not the sequenceNumber
+        const clientID = reportAction.clientID;
+        optimisticReportActions[clientID] = {
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
+            previousMessage: reportAction.message,
+            message: deletedMessage,
+            isDeletedPendingAdd,
```

Comment:
This adds a flag to hide the optimistic report action

To be honest we may not need the first 3 fields in this object:

```suggestion
            isDeletedPendingAdd,
```